pallets/allocations/src/lib.rs

```diff
@@ -107,6 +183,28 @@ pub mod pallet {
 
 	#[pallet::hooks]
 	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		// TODO benchmark on_initialize
+		fn on_initialize(n: BlockNumberFor<T>) -> Weight {
```

Comment:
Do we need to run this code at the beginning of every block, or is there an optimization possible here?

Also, what's the reason between using on_initialize VS on_finalize? Are there any special cases we should be aware of?

Reply:
The overhead on every block is very trivial. I have already optimized it to return quickly if nothing needs to be done. For our case it wouldn't matter whether we top up the quota on_initialize or on_finalize.